1_The_Physical/Guess_Point_Level_1_Stateless/1_Red_Green_Refactor/1_1_FizzbuzzV.2/src/index.spec.ts

```diff
@@ -8,5 +8,15 @@ describe('fizzbuzz', () => {
         it("returns '2' when given 2", () => {
         expect(fizzbuzz(2)).toBe("2")
         })
+        it("returns '100' when given 100", () => {
+            expect(fizzbuzz(100)).toBe("100")
+        })
     });
+    describe('gracefully handles out-of-bound numbers below 1 and above 100', () => {
+        it("returns 'Error' with message: 'Input must be a number between 1 and 100 inclusive', when given 0", () => {
```

Comment:
This test will break if you update the error message later. How to remove flakiness with error messages?

Reply:
The principle of Abstraction.

Separate the _what_ from the _how_.

Instead of saying that it returns 'Input must be a number between 1 and 100 inclusive' explicitly, back up and reflect on the actual **Responsibility** being expressed here.

It's a doing or knowing.

What does it do?

- it merely returns an error when the number is out of range.

Instead of testing against a string, test against a type.

```typescript
throw new OutOfRange('message here')
```

```typescript
test("throws an error when out of range", () => {
  ....
  expect(palindrome(0)).toThrow(OutOfRange);
});
```

This is also known as _polymorphism_, but _polymorphism_ is a form of _abstraction_, and _abstraction_ is the root principle of design.